themes/finna2/templates/Helpers/R2RestrictedRecordRegister.phtml

```diff
@@ -0,0 +1,27 @@
+<!-- START of: finna - Helpers/R2RestrictedRecordRegister.phtml -->
+<?php if ($this->weakLogin): ?>
+  <div class="r2-restricted-record-register alert alert-danger" role="alert">
+    <p><strong><?= $this->translate('R2_register_weak_loginmethod') ?></strong></p>
+  </div>
+<?php elseif (!$this->user && $this->id): ?>
+  <?php
+    // Register prompt on record page
+    $script = '$(document).ready(function() { finna.R2.initModal(); });';
+    $this->headScript()->appendScript($script);
+    // Explicitly redirect Collection page login to Record controller
+    $registerFormUrl = $this->url('r2record', ['id' => $this->id])
```

Comment:
Anna muutkin parametrit url-helperille (kolmas parametri: `['query' => ['login' => 'true', 'recordId' => $this->id, ...]]`)